### PR TITLE
support global shuffling for image data

### PR DESCRIPTION
**Pull request by gengyifeng**
Support global shuffling for image data with dmlc-core's PR.
To enable global shuffling, please add params "shuffle=True, shuffle_chunk_size = xxx"
in mx.io.ImageRecordIter for training.

### CODE REVIEW

File: src/io/iter_image_recordio.cc [hunk 1]
```diff
@@ -212,8 +221,32 @@ inline void ImageRecordIOParser::Init(
...
+                    "smaller chunk size";
+    }
+    if (param_.shuffle_chunk_size < 4) {
+      LOG(INFO) << "Chunk size: " << param_.shuffle_chunk_size
+                 << " MB which is less than 4 MB, please set "
+                    "larger chunk size";
+    }
+    // 1.1 ratio is for a bit more shuffle parts to avoid boundary issue
+    unsigned num_shuffle_parts =
+        std::ceil(source_->GetTotalSize() * 1.1 /
```
Comment (piiswrong) [inline src/io/iter_image_recordio.cc]: Does this guarantee correctness for very small/large size?

Comment (gengyifeng) [inline src/io/iter_image_recordio.cc]: for small size, num_shuffle_parts = 1, it uses the split directly, so there is no problem. for large size, the correctness is also ok, but too many chunks may slow down the data loading.

**Comment by gengyifeng** [comment]
It is mergable.

**Comment by piiswrong** [comment]
yes please open a pr

**Comment by TccccD** [comment]
@piiswrong @gengyifeng 
hello, I have a question, How does shuffle execute in distributed work, is it possible to shuffle the entire data set in every epoch and then assign it to each worker?  thanks.

**Pull request merged**